packages/compiler/test/shadow_css/keyframes_spec.ts

```diff
@@ -0,0 +1,510 @@
+        @keyframes baz {}
+        @keyframes quux {}
+        @keyframes grault {}
+        @keyframes waldo {}`;
+    const result = s(css, 'host-a');
+    const animationNames = [
+      'host-a_foo',
+      ' bar',
+      'host-a_baz',
+      ' qux ',
+      ' host-a_quux ',
+      'corge ',
+      'host-a_grault ',
+      'garply',
+      ' host-a_waldo',
+    ];
+    const expected = `animation-name: ${animationNames.join(',')};`;
+    expect(result).toContain(expected);
+  });
+
+  it('should handle (scope or not) multiple animation-names defined over multiple lines', () => {
+    const css = `
+        button {
+            animation-name: foo,
+                            bar,baz,
+                            qux ,
+                            quux ,
+                            grault,
+                            garply, waldo;
+        }
+        @keyframes foo {}
+        @keyframes baz {}
+        @keyframes quux {}
+        @keyframes grault {}`;
+    const result = s(css, 'host-a');
+    ['foo', 'baz', 'quux', 'grault'].forEach(
+        scoped => expect(result).toContain(`host-a_${scoped}`));
+    ['bar', 'qux', 'garply', 'waldo'].forEach(nonScoped => {
+      expect(result).toContain(nonScoped);
+      expect(result).not.toContain(`host-a_${nonScoped}`);
+    });
+  });
+
+  it('should handle (scope or not) multiple animation definitions in a single declaration', () => {
+    const css = `
+        div {
+            animation: 1s ease foo, 2s bar infinite, forwards baz 3s;
+        }
+
+        p {
+            animation: 1s "foo", 2s "bar";
+        }
+
+        span {
+            animation: .5s ease 'quux',
+                        1s foo infinite, forwards "baz'" 1.5s,
+                        2s bar;
+        }
+
+        button {
+            animation: .5s bar,
+                        1s foo 0.3s, 2s quux;
+        }
+
+        @keyframes bar {}
+        @keyframes quux {}
+        @keyframes "baz'" {}`;
+    const result = s(css, 'host-a');
+    expect(result).toContain('animation: 1s ease foo, 2s host-a_bar infinite, forwards baz 3s;');
+    expect(result).toContain('animation: 1s "foo", 2s "host-a_bar";');
+    expect(result).toContain(`
+            animation: .5s host-a_bar,
+                        1s foo 0.3s, 2s host-a_quux;`);
+    expect(result).toContain(`
+            animation: .5s ease 'host-a_quux',
+                        1s foo infinite, forwards "host-a_baz'" 1.5s,
+                        2s host-a_bar;`);
+  });
+
+  it(`should not modify css variables ending with 'animation' even if they reference a local keyframes`,
+     () => {
+       const css = `
+        button {
+            --variable-animation: foo;
+        }
+        @keyframes foo {}`;
+       const result = s(css, 'host-a');
+       expect(result).toContain('--variable-animation: foo;');
+     });
+
+  it(`should not modify css variables ending with 'animation-name' even if they reference a local keyframes`,
```

Comment:
```suggestion
  it(`should not modify css variables ending with 'animation-name' even if they reference a local keyframe`,
```

but I notice now that it's called `@keyframes`, so perhaps going with plural is more appropriate?

```suggestion
  it(`should not modify css variables ending with 'animation-name' even if they reference local keyframes`,
```

Reply:
mh.... I must admit that this is a bit confusing to me....

for example [mdn](https://developer.mozilla.org/en-US/docs/Web/CSS/@keyframes) calls the declaration "keyframe list" "keyframe rule" and the like

and a single keyframe is the... well... actual keyframe in the rule

as in:
![Screenshot at 2021-09-14 21-26-34](https://user-images.githubusercontent.com/61631103/133328999-7f7d78c6-4905-4cbb-bea0-64d2756b6971.png)

so I must admit this does confuses me.... maybe I we could call it a "keyframes identifier" or something like that?

(
as in
```
it(`should not modify css variables ending with 'animation-name' even if they reference a local keyframes identifier`,
```
)